pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -505,10 +502,8 @@ def visit_module(self, node: astroid.Module) -> None:
         """Called when a Module node is visited."""
         self._function_matchers = []
         for key, pattern in _MODULE_FILTERS.items():
-            if pattern.match(node.name):
-                if matches := _FUNCTION_MATCH.get(key):
-                    self._function_matchers = matches
-                break
+            if pattern.match(node.name) and (matches := _FUNCTION_MATCH.get(key)):
+                self._function_matchers.extend(matches)
```

Comment:
You mentioned that `any_platforms` is the one pattern which is responsible for overloads. Would it make sense to keep the basic structure (with `break`) in place and run the single pattern unconditionally?

That way we could at least preserve some performance improvements.

Reply:
I've moved to a single REGEX with a capturing group for the platform.
I've also added some tests to ensure the patterns match correctly.